### PR TITLE
WIP: suite-desktop deploy to data.trezor.io and some ci updates

### PR DESCRIPTION
**Pull request by vdovhanych**
- chore(ci): move autoupdate test deploy to deploy stage
- feat(ci): deploy suite-desktop to data.trezor.io

### CODE REVIEW

File: ci/deploy.yml [hunk 1]
```diff
@@ -226,3 +226,23 @@ components storybook deploy dev manual:
...
+  needs:
+    - suite-desktop build mac codesign
+    - suite-desktop build linux codesign
+    - suite-desktop build windows codesign
+  only:
+    refs:
+      - codesign
+  when: manual
+  variables:
+    DEPLOY_DIRECTORY: ${DEPLOY_BASE_DIR}/suite-desktop/${CI_BUILD_REF_NAME}
```
Comment (tsusanka) [inline ci/deploy.yml:242]: Should it go to suite-desktop/codesign?

Comment (tsusanka) [reply ci/deploy.yml:242]: Oh or that's what `${CI_BUILD_REF_NAME}` actually is isn't it?

**Review by tsusanka — APPROVED** [review]
LGTM! Will it work? 🙂

File: ci/deploy.yml (hunk 1 above)
Comment (vdovhanych) [reply ci/deploy.yml:242]: This job was already there. I just moved it to deploy.yml from release.yml. `${CI_BUILD_REF_NAME}` is `codesign` in this case, and it was deployed there a couple of times.

**Comment by vdovhanych** [comment]
> LGTM! Will it work? 🙂

It should but i want to test it before we merge it :) So ill stick with `WIP` for now

**Comment by tsusanka** [comment]
Ok ping me pls, when I should re-review.

**Comment by matejkriz** [comment]
I don't see the way how to use this in our auto-update flow. We can't check the new releases on Github and then download just the binaries from s3. We need to be able to check the releases directly on the s3 too. So we need also something like https://data.trezor.io/suite/releases/latest to be able to check for update on stable URL. 

And we need to think about where we're going to get the release notes. So far we are reading them from Github. But we could avoid using Github for auto-update completely if we put release notes directly in latest-xx.yml files (but then it's not dynamic and so easy to edit).

**Comment by tsusanka** [comment]
@vdovhanych now that I think about it can't we use cdn.trezor.io and not do this thing at all? Maybe we have discussed this already and I do not remember, but why not just cache responses to GitHub on the CDN similarly as we do with coin prices?

And instead of Github, Suite will contact cdn.trezor.io which will return cached response from GitHub.

**Comment by vdovhanych** [comment]
> @vdovhanych now that I think about it can't we use cdn.trezor.io and not do this thing at all? Maybe we have discussed this already and I do not remember, but why not just cache responses to GitHub on the CDN similarly as we do with coin prices?
> 
> And instead of Github, Suite will contact cdn.trezor.io which will return cached response from GitHub.

So I managed to proxy GitHub releases with CloudFront meaning every request through this(temporary for now) [domain](https://dcvi8zn1vmizo.cloudfront.net/trezor/trezor-suite/releases/) will be cached and served from amazon to customers. This way we only need to change the root URL for auto-updater and it should find all the files and all the paths that are present on Github now. @matejkriz is not sure about this solution and if it will be possible.

Another option is to upload it to our storage and also provide it through CloudFront this way we would need to have something like 
`/latest` - files here would always be overwritten with the latest stable release
`/early-access` - files here would also be overwritten 
in addition to the paths above we would also need to put the apps in their respectful version folders on the s3 bucket. so each release there would be 3 upload tasks. 
This is more time-consuming and we would also need to change quite a bit more stuff.

**Comment by tsusanka** [comment]
Ok, let's not do it this wait (yet). Let's assess priority of this first properly in https://github.com/trezor/trezor-suite/issues/4692 and see what options there are then. Sorry for the fuzz!

**Pull request closed without merging**